### PR TITLE
Adds SocketIOClient#once(event, callback)

### PR DESCRIPTION
**Pull request by KevinMartin**
When the specified event is emitted, the handler deregisters itself.

### CODE REVIEW

File: SocketIOClientSwift/SocketIOClient.swift [hunk 1]
```diff
@@ -425,6 +425,34 @@ public final class SocketIOClient: NSObject, SocketEngineClient {
     }
     
     /**
+    Adds a single-use handler for an event.
+    */
+    public func once(event: String, callback: NormalCallback) {
+        Logger.log("Adding once handler for event: %@", type: logType, args: event)
+        
+        let handler = SocketEventHandler(event: event) { data: NSArray?, ack: AckEmitter? in
```
Comment (nuclearace) [inline SocketIOClientSwift/SocketIOClient.swift]: You shouldn't need type declarations here. 

```swift
let handler = SocketEventHandler(event: event) {data, ack in
```

Should be fine

File: SocketIOClientSwift/SocketIOClient.swift [hunk 2]
```diff
@@ -425,6 +425,34 @@ public final class SocketIOClient: NSObject, SocketEngineClient {
...
+            handlers = ContiguousArray(handlers.filter { $0 != handler })
```
Comment (nuclearace) [inline SocketIOClientSwift/SocketIOClient.swift]: Shouldn't this be a weak reference?

File: SocketIOClientSwift/SocketIOClient.swift [hunk 3]
```diff
@@ -425,6 +425,34 @@ public final class SocketIOClient: NSObject, SocketEngineClient {
...
+        handlers.append(handler)
+    }
+    
+    /**
+    Adds a single-use handler for an event.
+    */
+    public func once(event event: String, callback: NormalCallbackObjectiveC) {
+        Logger.log("Adding once handler for event: %@", type: logType, args: event)
+        
+        let handler = SocketEventHandler(event: event) { data: NSArray?, ack: AckEmitterObjectiveC? in
```
Comment (nuclearace) [inline SocketIOClientSwift/SocketIOClient.swift]: Seems to be a build error here

File: SocketIOClientSwift/SocketIOClient.swift (hunk 2 above)
Comment (NachoSoto) [inline SocketIOClientSwift/SocketIOClient.swift]: No, `$0` won't outlive this declaration so there's no need to not make it `strong`.

Comment (nuclearace) [inline SocketIOClientSwift/SocketIOClient.swift]: I'm talking about referencing self.handlers inside of this entire block

**Comment by KevinMartin** [comment]
All fixed! :+1:

File: SocketIOClientSwift/SocketEventHandler.swift [hunk 4]
```diff
@@ -35,6 +35,7 @@ private func emitAckCallbackObjectiveC(socket: SocketIOClient?, num: Int?)
 }
 
 struct SocketEventHandler {
+    let id = NSUUID()
```
Comment (nuclearace) [inline SocketIOClientSwift/SocketEventHandler.swift]: Why is this needed?

Comment (KevinMartin) [inline SocketIOClientSwift/SocketEventHandler.swift]: You can't compare structs. I looked at other Swift-based event dispatchers that support removing specific handlers and they use id's like this to compare them.

**Comment by lightsprint09** [comment]
It would be great to cover this functions with tests... In my opinion every new functions should be tested, before the PR is merged
